ray-on-gke/tpu/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -1334,60 +1201,57 @@ func Test_MutatePod(t *testing.T) {
 	}{
 		"mutatePod missing cluster label": {
 			// missing Ray cluster label - returns error
-			testPod:             testCPUWorker.DeepCopy(),
+			testPod:             getTestCPUWorker("test-cluster", "test-group", "test-namespace"),
 			missingClusterLabel: true,
-			expectedError:       errors.New("Kuberay Pod missing RayCluster label"),
+			expectedError:       errors.New("Ray Pod created by KubeRay missing RayCluster label"),
 		},
 		"mutatePod missing container": {
 			// missing containers - returns error
-			testPod:             testCPUWorker.DeepCopy(),
+			testPod:             getTestCPUWorker("test-cluster", "test-group", "test-namespace"),
 			missingClusterLabel: false,
 			missingContainers:   true,
 			expectedError:       errors.New("Container path not specified"),
 		},
 		"mutatePod missing gke-tpu-topology nodeSelector": {
 			// requests TPUs, topology not specified - returns error
-			testPod:             testTPUWorker.DeepCopy(),
+			testPod:             getTestTPUWorker("test-cluster", "test-group", "test-namespace", "tpu-v4-podslice", "2x2x1"),
 			missingClusterLabel: false,
 			missingContainers:   false,
 			expectedTopology:    "",
-			expectedError:       errors.New("TPU topology not specified"),
+			expectedError:       errors.New("Ray Pod created by KubeRay missing TPU topology nodeSelector"),
 		},
 		"mutatePod in single-host TPU worker group": {
 			// requests TPUs, single-host - injects TPU_WORKER_ID, TPU_NAME and replicaIndex label
-			testPod:              testTPUWorker.DeepCopy(),
+			testPod:              getTestTPUWorker("test-cluster", "test-group", "test-namespace", "tpu-v4-podslice", "2x2x1"),
 			numOfHosts:           1,
 			expectedTopology:     "2x2x1",
 			expectedTPUChips:     resource.MustParse("4"),
 			expectedWorkerID:     "0",
 			expectedReplicaID:    0,
-			expectedWorkerName:   fmt.Sprintf("%s-%d", groupNameStr, 0),
-			expectedReplicaLabel: fmt.Sprintf("%s-%d", groupNameStr, 0),
+			expectedWorkerName:   fmt.Sprintf("%s-%d", "test-group", 0),
+			expectedReplicaLabel: fmt.Sprintf("%s-%d", "test-group", 0),
 		},
 		"mutatePod in multi-host TPU worker group": {
 			// requests TPUs, multi-host - injects hostname, subdomain, TPU_WORKER_ID, TPU_NAME,
 			// TPU_HOSTNAMES, a podAffinity field, and the replicaIndex label
-			testPod:            testTPUWorker.DeepCopy(),
+			testPod:            getTestTPUWorker("test-cluster", "test-group", "test-namespace", "tpu-v4-podslice", "2x2x4"),
 			numOfHosts:         4,
 			expectedTopology:   "2x2x4",
 			expectedTPUChips:   resource.MustParse("4"),
 			expectedWorkerID:   "0",
```

Comment:
Can we add test cases where the expected worker ID and replica ID is > 0?

Reply:
Done in [78e15a8](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/78e15a864ce8797d6c3ae2f3acbf0f2c2ba26751).